### ISSUE TITLE
Training Mini-batches of data (without labels) for unsupervised learning problem

### ISSUE THREAD

**Issue by vishalbhalla** [tensorflow/tensorflow#13262]
Hi Everyone,

Has anyone trained mini-batches of data for an unsupervised learning problem? 
The feed_dict uses the label and in an unsupervised setting. How do you overcome that? Could we use fake labels that never contribute to the loss function?

Basically, I want to iterate over my huge dataset and then optimize a custom loss function. However, I couldn't figure out how to retain my training parameters (weights) when using a new mini-batch from the data explicitly.

For example, the whole dataset is 6000 points and the mini-batch size is 600.
Currently, for every mini-batch I could only use new independent weight parameters because the weights are initialized based on the data points from this mini-batch. When we optimize the loss over the first mini-batch of 600 data points, we get some optimized weights. How does one use these weights to optimize the next mini-batch of 600 data points and so on. The problem is we cannot use a shared global variable.

Any help or pointers in this regard would be really appreciated. Thanks in advance!

**Comment by aadamson** [comment]
Variables should maintain state between calls to `run` so unless you're reinitializing the weights between every minibatch, you shouldn't have an issue. An example demonstrating your problem would be helpful.

Also, his question would be better suited for Stack Overflow than a GH issue

**Comment by vishalbhalla** [comment]
Hi Adam,

I researched on stackoverflow forums but couldn't find anything relevant for mini-batches over unsupervised data. Hence, I posted here after a friend's suggestion. My apologies for the same.

'f' is my whole dataset say text data of N points with dimension D
U is cluster centroid with K clusters again of dimension D

I define my variables as below:
```
F = tf.Variable(f.astype(np.float32), name='F') 
U = tf.Variable(u.astype(np.float32), name='U')
FMod = tf.reshape(F, [N/K, K, D], name='FMod')
UMod = tf.reshape(U, [1, K, D], name='UMod')
```

Then I define a custom loss or objective function as 'objective'

Next I use an optimizer
```
optimizer = tf.train.AdamOptimizer(learning_rate)
train_W = optimizer.minimize(objective, var_list=[F, U])
```

Finally, I evaluate the variable as
```
with tf.Session() as sess:
                
    # Initialize all the variables
    sess.run(init_op)

    for n in range(noEpochs):

        objval1 = sess.run([train_W, objective])
		
```

The thing I am stuck at - is to iterate over batches of my data 'f' which is ultimately used in the optimizer train_W. If I have a for loop over these mini-batches, I will assign a new variable train_W for each of these iterations. How can I pass this value so that it can be used in the next mini-batch?
 
The full code can be found at this link: [Semi-supervised Clustering](https://github.com/vishalbhalla/semi-supervised-clustering/blob/master/semi-supervised-clustering/code/Semi-supervised%20Clustering.ipynb)

**Comment by asimshankar** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
